tests/x509_openssl_ut/x509_openssl_ut.c

```diff
@@ -117,10 +117,12 @@ MOCKABLE_FUNCTION(, long, SSL_CTX_ctrl, SSL_CTX*, ctx, int, cmd, long, larg, voi
 MOCKABLE_FUNCTION(, unsigned long, ERR_peek_last_error);
 MOCKABLE_FUNCTION(, void, ERR_clear_error);
 
+#ifndef OPENSSL_NO_ENGINE
```

Comment:
Can't add a comment there but I think you should add a `#ifndef` on line  37 (`#include "openssl/engine.h"`) to fix the OSX bug.

Reply:
aha! that would make sense, thanks. i'll try and get back to this soon 👍🏻

Reply:
I've guarded that include now, hopefully everything is all good now 🙏